python/tests/test_transforms.py

```diff
@@ -0,0 +1,526 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import datetime
+from decimal import Decimal, getcontext
+from uuid import UUID
+
+import pytest
+
+from iceberg import transforms
+from iceberg.transforms import UnknownTransform
+from iceberg.types import (
+    BinaryType,
+    BooleanType,
+    DateType,
+    DecimalType,
+    DoubleType,
+    FixedType,
+    FloatType,
+    IntegerType,
+    ListType,
+    LongType,
+    MapType,
+    NestedField,
+    StringType,
+    StructType,
+    TimestampType,
+    TimestamptzType,
+    TimeType,
+    UUIDType,
+)
+
+
+@pytest.mark.parametrize(
+    "test_input,test_type,expected",
+    [
+        (1, IntegerType, 1392991556),
+        (34, IntegerType, 2017239379),
+        (34, LongType, 2017239379),
+        (1, FloatType, -142385009),
+        (1, DoubleType, -142385009),
+        (17486, DateType, -653330422),
+        (81068000000, TimeType, -662762989),
+        (
+            int(
+                datetime.fromisoformat("2017-11-16T22:31:08+00:00").timestamp()
+                * 1000000
+            ),
+            TimestampType,
+            -2047944441,
+        ),
+        (
+            int(
+                datetime.fromisoformat("2017-11-16T14:31:08-08:00").timestamp()
+                * 1000000
+            ),
+            TimestamptzType,
+            -2047944441,
+        ),
+    ],
+)
+def test_spec_values_int(test_input, test_type, expected):
+    assert transforms.Bucket._FUNCTIONS_MAP[test_type][0](test_input) == expected
```

Comment:
What about the fixed and binary cases?

Reply:
Updated with fixed, binary, string, and uuid.